src/XMLSecurityDSig.php

```diff
@@ -237,7 +246,7 @@ public function setCanonicalMethod($method)
             if ($sinfo = $nodeset->item(0)) {
                 $query = './'.$this->searchpfx.'CanonicalizationMethod';
                 $nodeset = $xpath->query($query, $sinfo);
-                if (! ($canonNode = $nodeset->item(0))) {
+                if (! ($canonNode = $nodeset->item(0)) || !$canonNode instanceof DOMElement) {
```

Comment:
Why the test for DOMElement as the XPath query would only return either an empty set or elements named CanonicalizationMethod. It does tho appear it illustrates a bug tho as the query should be $query = './'.$this->searchpfx.':CanonicalizationMethod'; Note that the ":" in front of CanonicalizationMethod is missing so this query never properly worked anyways

Reply:
You are absolutely right. The reason for the test is that psalm, the static analyzer, can't correlate the actual query (a string) with the possible results obtained out from `query()`. So the test was trying to avoid psalm complaining about it, but probably the most correct way to fix it is actually to tell it that the result is guaranteed to be a `DOMNodeList` containing only instances of `DOMElement` (if anything at all).

I've fixed that, as well as the broken query, and included psalm as part of the build process. It should help us to avoid introducing bugs and mistakes into the code base from now on.